thenewboston_node/blockchain/views/blockchain_states_meta.py

```diff
@@ -0,0 +1,91 @@
+from urllib.parse import urljoin
+
+from django.conf import settings
+
+from drf_spectacular.utils import OpenApiParameter, extend_schema
+from rest_framework.exceptions import APIException, ParseError
+from rest_framework.viewsets import ReadOnlyModelViewSet
+
+from thenewboston_node.blockchain import exceptions
+from thenewboston_node.blockchain.pagination import CustomLimitOffsetPagination
+from thenewboston_node.blockchain.serializers.blockchain_states_meta import BlockchainStatesMetaSerializer
+from thenewboston_node.business_logic.blockchain.base import BlockchainBase
+from thenewboston_node.business_logic.blockchain.file_blockchain import FileBlockchain
+from thenewboston_node.business_logic.models import BlockchainState
+from thenewboston_node.business_logic.node import get_node_identifier
+
+
+class BlockchainStatesMetaViewSet(ReadOnlyModelViewSet):
+    serializer_class = BlockchainStatesMetaSerializer
+    pagination_class = CustomLimitOffsetPagination
+
+    @extend_schema(parameters=[OpenApiParameter('id', location=OpenApiParameter.PATH, description='Block number')])
+    def retrieve(self, request, *args, **kwargs):
+        return super().retrieve(request, *args, **kwargs)
+
+    @extend_schema(
+        parameters=[
+            OpenApiParameter('ordering', location=OpenApiParameter.QUERY, description='asc/desc', default='asc'),
+        ]
+    )
+    def list(self, request):  # noqa: A003
+        return super().list(request)
+
+    def get_queryset(self):
+        blockchain = self._get_blockchain()
+        blockchain_states = self._get_blockchain_states(blockchain)
+        return [
+            self._make_blockchain_state_meta(blockchain, blockchain_state) for blockchain_state in blockchain_states
```

Comment:
@dmugtasimov This new design is very suboptimal. Now on every list request all blockchain states are read and deserialized. Maybe add `only_meta=True` argument to `yield_blockchain_states`?

Reply:
@andbortnik leave it for now. we will optimize it later. I think it is even cached already.

But we need it properly optimized regarding to `limit` get parameter. We should not read all blockchain states to get only one.

Reply:
> But we need it properly optimized regarding to limit get parameter.

This is how pagination is implemented. It requires either `queryset` or `list` to get object count.